vimode/src/cmds/fold.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 Sylvain Cresto <scresto@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "cmds/fold.h"
+#include "utils.h"
+
+/* fold command does not depend on state of parents */
+#define FOLD_NONE		0
+/* fold command depend on state of parents */
+#define FOLD_PARENT		1
+/* fold command depend on state of contracted parent if it exists */
+#define FOLD_CONTRACTED		2
```

Comment:
I was super-confused by the naming. I's suggest the following:
* `FOLD_NONE` -> `GOTO_NEAREST_PARENT`
* `FOLD_PARENT` -> `GOTO_TOPMOST_PARENT`
* `FOLD_CONTRACTED` -> `GOTO_CONTRACTED_PARENT`

plus the corresponding  comment changes.